src/openvic-simulation/economy/production/ResourceGatheringOperation.cpp

```diff
@@ -14,8 +24,228 @@ ResourceGatheringOperation::ResourceGatheringOperation(
 	output_quantity_yesterday { new_output_quantity_yesterday },
 	unsold_quantity_yesterday { new_unsold_quantity_yesterday },
 	size_multiplier { new_size_multiplier },
-	employees { std::move(new_employees) } {}
+	employees { std::move(new_employees) },
+	total_employees_count_cache {},
+	total_paid_employees_count_cache { 0},
+	total_owner_income_cache {},
+	total_employee_income_cache {}
+{}
 
-ResourceGatheringOperation::ResourceGatheringOperation(
-	ProductionType const& new_production_type, fixed_point_t new_size_multiplier
-) : ResourceGatheringOperation { new_production_type, new_size_multiplier, 0, 0, 0, {} } {}
+ResourceGatheringOperation::ResourceGatheringOperation() : ResourceGatheringOperation {
+	nullptr, fixed_point_t::_0(),
+	 fixed_point_t::_0(), fixed_point_t::_0()
+	 , fixed_point_t::_0(), {}
+} {}
+
+void ResourceGatheringOperation::initialise_for_new_game(ProvinceInstance& location) {
+	if(this->production_type == nullptr) {
+		output_quantity_yesterday = 0;
+		revenue_yesterday = 0;
+		return;
+	}
+
+	ProductionType const& production_type = *(this->production_type);
+
+	const Pop::pop_size_t total_worker_count_in_province = update_size_and_return_total_worker_count(location);
+	hire(location, total_worker_count_in_province);
+	output_quantity_yesterday = produce(location);
+	revenue_yesterday = output_quantity_yesterday * production_type.get_output_good().get_base_price(); //TODO sell on market
+	pay_employees(location, revenue_yesterday, total_worker_count_in_province);	
+}
+
+Pop::pop_size_t ResourceGatheringOperation::update_size_and_return_total_worker_count(ProvinceInstance& location) {
+	Pop::pop_size_t total_worker_count_in_province = 0; //not counting equivalents
+	if(production_type == nullptr) {
+		size_multiplier = fixed_point_t::_0();
+	}
+	else {
+		ProductionType const& production_type = *(this->production_type);
+		std::vector<Job> const& jobs = production_type.get_jobs();
+		//can't use pop_type_distribution as it is not filled correctly yet (possibly due to equivalent pop type conversion)
+		for (Pop const& pop : location.get_pops()){
+			PopType const* pop_type = pop.get_type();
+			for(Job const& job : jobs) {
+				if(job.get_pop_type() == pop_type) {
+					total_worker_count_in_province += pop.get_size();
+					break;
+				}
+			}
+		}
+
+		const fixed_point_t base_workforce_size = production_type.get_base_workforce_size();
+		TerrainType const* const terrain_type = location.get_terrain_type();
+		if(terrain_type != nullptr) {
+			ModifierValue const& terrain_modifiers = terrain_type->get_modifier();
+			if(production_type.is_farm()) {
+				//base_workforce_size *= terrain_modifiers["farm_rgo_size"];
+			}
+			if(production_type.is_mine()) {
+				//base_workforce_size *= terrain_modifiers["mine_rgo_size"];
+			}
+		}
```

Comment:
When I've finished implementing the resultant modifier system you won't need to go via the terrain type, the province will have a `ModiferSum` variable which you can look up the farm or mine size modifier from.

Reply:
Only terrain (& continent?) is used for rgo size calculation. Country modifiers, province modifiers, tech, issues, etc are not used iirc.